tests/jq.test

```diff
@@ -1814,3 +1814,35 @@ true
 tojson | fromjson
 {"a":nan}
 {"a":null}
+
+pick(.a.b.c)
+null
+{"a":{"b":{"c":null}}}
+
+pick(first)
+[1,2]
+[1]
+
+pick(first|first)
+[[10,20],30]
+[[10]]
+
+pick(last)
+[[10,20],30]
+[null,30]
+
+[inputs]
+null
+[]
+
+[input?]
+null
+[]
```

Comment:
Calling `input/1` in `--run-tests` mode causes a segmentation fault which we should probably fix, but I am not sure how you expect those tests to work

Reply:
I ran `--run-tests` without incident on a Mac.  So maybe there is a problem that needs to be fixed on some platforms?

Reply:
@pkoppstein The issue is that the struct field that specifies the input used by `input/1` is unitialised when running `--run-tests`. It could be that your system's implementation of `malloc()` returns values initialised to `0`, or something else, and those uninitaliased fields happened to be set to `0`, so it worked.
On GNU libc systems, `malloc()` typically returns memory initialised with `0x61` bytes for debug builds to help catch those problems, so the `jq->input_cb` field was set to `0x6161616161616161`, and, since it is not `NULL`, `input/1` tried to call it, triggering a segfault.
Anyway, anything could have happened.